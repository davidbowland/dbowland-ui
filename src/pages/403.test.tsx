import '@testing-library/jest-dom'
import React from 'react'
import { mocked } from 'jest-mock'
import { render } from '@testing-library/react'

import Forbidden from './403'
import ServerErrorMessage from '@components/server-error-message'
import Themed from '@components/themed'
import TitleBar from '@components/title-bar'

jest.mock('@aws-amplify/analytics')
jest.mock('@components/server-error-message')
jest.mock('@components/themed')
jest.mock('@components/title-bar')

describe('403 error page', () => {
  beforeAll(() => {
    mocked(ServerErrorMessage).mockReturnValue(<></>)
    mocked(Themed).mockImplementation(({ children }) => <>{children}</>)
    mocked(TitleBar).mockReturnValue(<></>)
  })

  test('expect rendering Forbidden renders ServerErrorMessage', () => {
    const expectedTitle = '403: Forbidden'
    render(<Forbidden />)
    expect(mocked(ServerErrorMessage)).toHaveBeenCalledWith(
      expect.objectContaining({ title: expectedTitle }),
      expect.anything()
    )
    expect(mocked(ServerErrorMessage)).toHaveBeenCalledTimes(1)
  })

  test('expect rendering Forbidden renders TitleBar', () => {
    render(<Forbidden />)
    expect(mocked(TitleBar)).toHaveBeenCalledTimes(1)
  })
})
