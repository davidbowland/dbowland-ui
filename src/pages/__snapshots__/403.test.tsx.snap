// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`403 error page Ensure snapshot match 1`] = `
<main
  class="sc-dkPtyc iwVFAu"
>
  <h1
    class="sc-gsDJrp gxPvwY"
  >
    403: Forbidden
  </h1>
  <p
    class="sc-bdvvaa zWSMl"
  >
    You are not allowed to access the resource you requested. If you feel you have reached this page in error, please contact the webmaster.
    <br />
    <br />
    <a
      href="/"
    >
      Go home
    </a>
    .
  </p>
</main>
`;
